substrate/client/service/src/builder.rs

```diff
@@ -349,7 +349,7 @@ where
 }
 
 /// Parameters to pass into `build`.
-pub struct SpawnTasksParams<'a, TBl: BlockT, TCl, TExPool, TRpc, Backend> {
+pub struct SpawnTasksParams<'a, TBl: BlockT, TCl, TExPool: ?Sized, TRpc, Backend> {
```

Comment:
All these `?Sized` are "confusing". IMO we should just make `transaction_pool: Arc<dyn TransactionPool>` and thats it.

Reply:
Do I understand correctly that you want to remove `TransactionPool` generic type parameter from all generic types?

Reply:
Yes, would probably make it easier.